tests/unit/pipeline/extractors/stores/test_splunk_extractor.py

```diff
@@ -0,0 +1,657 @@
+    """Sample Splunk search results in the format returned by the results API."""
+    return {
+        "results": [
+            {
+                "_time": "2023-01-01T10:00:00",
+                "host": "server1",
+                "message": "Login successful",
+            },
+            {
+                "_time": "2023-01-01T10:01:00",
+                "host": "server2",
+                "message": "Error occurred",
+            },
+            {"_time": "2023-01-01T10:02:00", "host": "server1", "message": "Logout"},
+        ]
+    }
+
+
+@pytest.fixture
+def splunk_extractor():
+    """Create a SplunkExtractor instance for testing."""
+    return SplunkExtractor.from_file_data(
+        base_url="https://splunk.example.com:8089",
+        query="index=test | head 10",
+        auth_token="test-token-123",
+        earliest_time="-1h",
+        latest_time="now",
+        chunk_size=100,
+    )
+
+
+@pytest.fixture
+def splunk_extractor_basic_auth():
+    """Create a SplunkExtractor instance with basic auth for testing."""
+    return SplunkExtractor.from_file_data(
+        base_url="https://splunk.example.com:8089",
+        query="index=test",
+        username="testuser",
+        password="testpass",
+    )
+
+
+def test_splunk_extractor_from_file_data_with_token_auth():
+    extractor = SplunkExtractor.from_file_data(
+        base_url="https://splunk.example.com:8089/",  # trailing slash should be stripped
+        query="index=main",
+        auth_token="my-token",
+        earliest_time="-24h",
+        chunk_size=500,
+    )
+
+    assert_that(extractor.base_url, equal_to("https://splunk.example.com:8089"))
+    assert_that(extractor.query, equal_to("index=main"))
+    assert_that(extractor.auth_token, equal_to("my-token"))
+    assert_that(extractor.username, equal_to(None))
+    assert_that(extractor.password, equal_to(None))
+    assert_that(extractor.earliest_time, equal_to("-24h"))
+    assert_that(extractor.chunk_size, equal_to(500))
+
+
+def test_splunk_extractor_from_file_data_with_basic_auth():
+    extractor = SplunkExtractor.from_file_data(
+        base_url="https://splunk.example.com:8089",
+        query="index=security",
+        username="admin",
+        password="secret",
+    )
+
+    assert_that(extractor.auth_token, equal_to(None))
+    assert_that(extractor.username, equal_to("admin"))
+    assert_that(extractor.password, equal_to("secret"))
+
+
+def test_splunk_extractor_from_file_data_defaults():
+    extractor = SplunkExtractor.from_file_data(
+        base_url="https://splunk.example.com:8089",
+        query="index=main",
+    )
+
+    assert_that(extractor.verify_ssl, equal_to(True))
+    assert_that(extractor.request_timeout_seconds, equal_to(300))
+    assert_that(extractor.chunk_size, equal_to(1000))
+
+
+# Helper property tests
+def test_splunk_extractor_auth_property_with_token(splunk_extractor):
+    assert_that(splunk_extractor._auth, equal_to(None))  # Token goes in header
```

Comment:
in general I would avoid testing internal/private properties and functions to avoid making tests that rely on implementation details.

These tests can be handled by `responses` and `pytest-httpx` by having these mock libraries watching for expected auth headers.